app/src/main/java/org/p2p/wallet/home/addmoney/ui/AddMoneyDialog.kt

```diff
@@ -0,0 +1,154 @@
+            AddMoneyDialog().show(fm, tag)
+        }
+    }
+
+    private val binding: DialogAddMoneyBinding by viewBinding()
+    private val receiveFragmentFactory: ReceiveFragmentFactory by inject()
+    private val strigaFragmentFactory: StrigaFragmentFactory by inject()
+
+    override val presenter: AddMoneyDialogContract.Presenter by inject()
+
+    private val cellAdapter = CommonAnyCellAdapter(
+        mainCellDelegate(inflateListener = {
+            it.setOnClickAction { _, item ->
+                presenter.onItemClick(item.typedPayload())
+            }
+        })
+    )
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_RoundedSnow
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            recyclerViewCells.disableScrolling()
+            recyclerViewCells.attachAdapter(cellAdapter)
+            recyclerViewCells.addItemDecoration(offsetFinanceBlockDecoration(8.toPx()))
+        }
+    }
+
+    override fun showItemProgress(itemType: AddMoneyItemType, showProgress: Boolean) {
+        cellAdapter.updateItem<MainCellModel>(
+            predicate = { it is MainCellModel && it.payload == itemType },
+            transform = {
+                val rightSideCellModel = if (showProgress) {
+                    RightSideCellModel.Progress(
+                        indeterminateProgressTint = R.color.night
+                    )
+                } else {
+                    RightSideCellModel.IconWrapper(
+                        iconWrapper = IconWrapperCellModel.SingleIcon(
+                            icon = ImageViewCellModel(
+                                icon = DrawableContainer(R.drawable.ic_chevron_right),
+                                iconTint = R.color.icons_mountain
+                            )
+                        )
+                    )
+                }
+                it.copy(rightSideCellModel = rightSideCellModel)
+            }
+        )
+    }
+
+    override fun navigateToBankTransferTarget(target: StrigaUserStatusDestination) {
+        strigaFragmentFactory.signupFlowFragment(target)?.let(::dismissAndNavigate)
+    }
+
+    override fun navigateToBankCard(tokenToBuy: Token, paymentMethod: PaymentMethod.MethodType) {
+        dismissAndNavigate(
+            NewBuyFragment.create(
+                token = tokenToBuy,
+                preselectedMethodType = paymentMethod
+            )
+        )
+    }
+
+    override fun navigateToKycPending() {
+        dismiss()
+        strigaFragmentFactory.showPendingBottomSheet(requireActivity().supportFragmentManager)
+    }
+
+    override fun setCellItems(items: List<AnyCellItem>) {
+        cellAdapter.items = items
+    }
+
+    override fun navigateToCrypto() {
+        dismissAndNavigate(receiveFragmentFactory.receiveFragment())
+    }
+
+    override fun showUiKitSnackBar(
```

Comment:
let's put some comment here why are we overriding default impl

Reply:
added comment with link to tech.debt task